CMakeLists.txt

```diff
@@ -1,28 +1,26 @@
 cmake_minimum_required(VERSION 3.15)
 
-# User is responsible to set two mandatory options:
-# FREERTOS_CONFIG_FILE_DIRECTORY
-# FREERTOS_PORT
+# User is responsible to one mandatory option:
+#   FREERTOS_PORT
+#
+# User is responsible for one library target:
+#   freertos_config ,typcially an INTERFACE library
 #
 # User can choose which heap implementation to use (either the implementations
 # included with FreeRTOS [1..5] or a custom implementation ) by providing the
 # option FREERTOS_HEAP. If the option is not set, the cmake will default to
 # using heap_4.c.
 
-# Absolute path to FreeRTOS config file directory
-set(FREERTOS_CONFIG_FILE_DIRECTORY "" CACHE STRING "Absolute path to the directory with FreeRTOSConfig.h")
-
-if(NOT FREERTOS_CONFIG_FILE_DIRECTORY)
-    message(FATAL_ERROR " FreeRTOSConfig.h file directory not specified. Please specify absolute path to it from top-level CMake file:\n"
-        "  set(FREERTOS_CONFIG_FILE_DIRECTORY <absolute path to FreeRTOSConfig.h directory> CACHE STRING \"\")\n"
-        " or from CMake command line option:\n"
-        "  -DFREERTOS_CONFIG_FILE_DIRECTORY='/absolute_path/to/FreeRTOSConfig.h/directory'")
-elseif(NOT EXISTS ${FREERTOS_CONFIG_FILE_DIRECTORY}/FreeRTOSConfig.h)
-    message(FATAL_ERROR " FreeRTOSConfig.h file not found in the directory specified (${FREERTOS_CONFIG_FILE_DIRECTORY})\n"
-        " Please specify absolute path to it from top-level CMake file:\n"
-        "  set(FREERTOS_CONFIG_FILE_DIRECTORY <absolute path to FreeRTOSConfig.h directory> CACHE STRING \"\")\n"
-        " or from CMake command line option:\n"
-        "  -DFREERTOS_CONFIG_FILE_DIRECTORY='/absolute_path/to/FreeRTOSConfig.h/directory'")
+# `freertos_config` target defines the path to FreeRTOSConfig.h and optionally other freertos based config files
```

Comment:
Is this a backward incompatible change? If so, is there a way to make it backward compatible?

Reply:
Yes - have added a warning if FREERTOS_CONFIG_FILE_DIRECTORY is used and if it is will add the appropriate info for freertos_config target if that has not been configured.